packages/taiko-client/driver/preconf_blocks/queue.go

```diff
@@ -60,20 +60,43 @@ func (q *payloadQueue) get(id uint64, hash common.Hash) *eth.ExecutionPayload {
 	return nil
 }
 
-// get retrieves a previously stored payload item or nil if it does not exist.
-func (q *payloadQueue) getChild(id uint64, hash common.Hash) *eth.ExecutionPayload {
+// getChildren retrieves the longest previously stored payload items that are children of the
+// given parent payload.
+func (q *payloadQueue) getChildren(parentID uint64, parentHash common.Hash) []*eth.ExecutionPayload {
 	q.lock.RLock()
 	defer q.lock.RUnlock()
 
-	for _, item := range q.payloads {
-		if item == nil {
-			return nil // no more items
+	longestChildren := []*eth.ExecutionPayload{}
+
+	var searchLongestChildren func(currentpayload *eth.ExecutionPayload, chain []*eth.ExecutionPayload)
+	searchLongestChildren = func(currentpayload *eth.ExecutionPayload, chain []*eth.ExecutionPayload) {
+		children := []*eth.ExecutionPayload{}
+		for _, item := range q.payloads {
+			if item == nil {
+				break // no more items
+			}
+			if item.id == uint64(currentpayload.BlockNumber)+1 && item.payload.ParentHash == currentpayload.BlockHash {
+				children = append(children, item.payload)
+			}
 		}
-		if item.id == id+1 && item.payload.ParentHash == hash {
-			return item.payload
+		if len(children) == 0 {
+			if len(chain) > len(longestChildren) {
+				longestChildren = append([]*eth.ExecutionPayload{}, chain...)
+			}
+			return
+		}
+
+		for _, child := range children {
+			searchLongestChildren(child, append(chain, child))
 		}
 	}
-	return nil
+
+	searchLongestChildren(&eth.ExecutionPayload{
+		BlockNumber: eth.Uint64Quantity(parentID),
+		BlockHash:   parentHash,
+	}, []*eth.ExecutionPayload{})
```

Comment:
```suggestion
	}, nil)
```

Reply:
Although they are the same, `[]*eth.ExecutionPayload{}` looks more readable to me, no?